### DISCUSSION TITLE
Why isn't react-aria as popular as other libraries/frameworks?

### DISCUSSION THREAD

**Discussion by timurmaio** [adobe/react-spectrum#4643] [General]
Why isn't react-aria as popular as other libraries/frameworks? I came across this link, https://npmtrends.com/@ariakit/react-vs-@headlessui/react-vs-@radix-ui/primitive-vs-react-aria-vs-react-aria-components, and it seems like react-aria is not getting much love as a low-level headless component library. I've also noticed that Radix is getting a lot more attention on Twitter. But I don't get it. To me, react-aria is a super complete, complex, battle-tested library from Adobe. Can someone explain why this is happening? Or maybe npmtrends isn't the best way to measure popularity?

**Comment by devongovett** [comment]
The `@react-aria/utils` package is a better measure than the `react-aria` monopackage since it is a dependency of pretty much all of our stuff. The monopackage was not well advertised in the docs until more recently so lots of folks are using individual packages instead. https://npmtrends.com/@ariakit/react-vs-@headlessui/react-vs-@radix-ui/primitive-vs-@react-aria/utils-vs-react-aria-vs-react-aria-components

As you can see it is a lot closer but still a bit lower. I think it's because other libraries offer higher level APIs than we have historically offered, which developers find attractive. That's one of the reasons we're working on react-aria-components.

**Reply by timurmaio** [reply]
Thank you for your response @devongovett! I'm pleasantly surprised to receive such a quick answer from you, considering the substantial workload you have as a maintainer of numerous open-source libraries. I'm a huge fan of your work and I believe it deserves more recognition. I consistently follow your blog posts, which highlight the incredible effort and time you and your team dedicate to projects like React-aria and others. While I have your attention, I wanted to take a moment to express my gratitude to you. Thank you so much!

**Comment by rolanday** [comment]
+1. If not for react-aria (and framer), I would exit react for truly reactive framework like qwik for solid. Have tried most of the other UI frameworks, which are like candy to a baby for how easy to get up and running, but always end up hitting a wall at some point where I spend more time working around the framework or overcoming limitations than the time saved. React-aria is first framework where gets out of my way, but has a steeper learning curve. Great UI framework as makes me feel like possibilities are endless. Thank you!!
